apps/server/src/controllers/integrationController.ts

```diff
@@ -222,6 +237,74 @@ export function dispatchFromAdapter(type: string, payload: unknown, source?: 'os
       return { topic: 'timer', payload: timer };
     }
 
+    // ontime/change/{eventID}/{propertyName}
+    case 'change': {
+      try {
+        if (params.length < 2) {
+          throw new Error('Invalid parameters');
+        }
+
+        const eventID = params[0];
+        const propertyName = params[1] as keyof OntimeEvent;
+
+        const isKeyOfOntimeEvent = (key: string): key is keyof OntimeEvent => {
+          return key in event;
+        };
+
+        if (!isKeyOfOntimeEvent(propertyName)) {
+          throw new Error(`Cannot update unknown event property ${propertyName}`);
+        }
+
+        const acceptedPropertyPayloadMap = {
+          title: 'string',
+          subtitle: 'string',
```

Comment:
what if the value here is a callback to validate as the correct type, like 


```ts
function maybeString(value: unknown) {
  if (value == null) {
      throw new Error("Value type invalid")
  }
  return String(value)
}


function maybeBoolean(value: unknown) {
  if (value == null) {
      throw new Error("Value type invalid")
  }
  return Boolean(value);
}


// should probably handling force ints too. all timers in ontime need to be ints
function maybeNumber(value: unknown) {
  if (value == null) {
      throw new Error("Value type invalid")
  }
  return Number(value);
}

const whitelistedPayload = {
  title: maybeString,
  subtitle: maybeString,
  presenter: maybeString,
  note: maybeString,
  cue: maybeString,

  duration: maybeNumber,

  isPublic: maybeBoolean,
  skip: maybeBoolean,

  colour: maybeString,
  user0: maybeString,
  user1: maybeString,
  user2: maybeString,
  user3: maybeString,
  user4: maybeString,
  user5: maybeString,
  user6: maybeString,
  user7: maybeString,
  user8: maybeString,
  user9: maybeString,
};
```

Reply:
maybe have it all this inside something like `integrationController.config.ts` then? then I can move a big chunk of the parsing and validation logic to it, maybe also have `isKeyOfOntimeEvent` there?

Reply:
I think a config file makes sense and it will be helpful to encapsulate business logic related to the integration (in my example, parse and whitelisted)

maybeNumber, maybeBoolean and maybeString are generic utilities that can go in the utils package. They should probably be called coerceBoolean coerceString and CoerceNumber instead

the isKeyOfOntimeEvent is more generic to the domain logic and makes more sense in the types package

Reply:
thanks, it really makes sense now, thanks for the explanation on the architecture, it is really helpful!